test/test.compute.ts

```diff
@@ -120,6 +118,44 @@ describe('Compute auth client', () => {
     });
   });
 
+  it('should retry calls to the metadata service if there are network errors',
+     (done) => {
+       const scopes = [
+         nock(HOST_ADDRESS).get(tokenPath).times(2).replyWithError({
+           code: 'ENOTFOUND'
+         }),
+         nock(HOST_ADDRESS).get(tokenPath).reply(200, {
+           access_token: 'abc123',
+           expires_in: 10000
+         })
```

Comment:
If I am reading `nock` documentation correctly, I think this can be simplified:

```js
const scope = nock(HOST_ADDRESS).get(tokenPath).times(2)
  .replyWithError({ code: 'ENOTFOUND' }),
  .reply(200, {
    access_token: 'abc123',
    expires_in: 10000
});
```

And then you can get rid of the `foreach` below.